src/plugins/dashboard/public/application/embeddable/dashboard_container.tsx

```diff
@@ -159,29 +159,55 @@ export class DashboardContainer extends Container<InheritedChildInput, Dashboard
         [placeholderPanelState.explicitInput.id]: placeholderPanelState,
       },
     });
-    newStateComplete.then((newPanelState: Partial<PanelState>) => {
-      const finalPanels = { ...this.input.panels };
-      delete finalPanels[placeholderPanelState.explicitInput.id];
-      const newPanelId = newPanelState.explicitInput?.id
-        ? newPanelState.explicitInput.id
-        : uuid.v4();
-      finalPanels[newPanelId] = {
-        ...placeholderPanelState,
-        ...newPanelState,
-        gridData: {
-          ...placeholderPanelState.gridData,
-          i: newPanelId,
-        },
+    newStateComplete.then((newPanelState: Partial<PanelState>) =>
+      this.replacePanel(placeholderPanelState, newPanelState)
+    );
+  }
+
+  public replacePanel(
```

Comment:
Just a small reminder - could this be extracted in a separate PR? It's not Lens related, so we'll be keeping both PRs smaller.

Reply:
Good call. Created https://github.com/elastic/kibana/pull/71130